sms_sponsorship/models/sms_child_request.py

```diff
@@ -83,10 +83,14 @@ def _compute_full_url(self):
     @api.depends('date')
     def _compute_event(self):
         for request in self.filtered('date'):
-            request.event_id = self.env['crm.event.compassion'].search([
+            event_id = self.env['crm.event.compassion'].search([
                 ('accepts_sms_booking', '=', True),
-                ('start_date', '<=', request.date)
-            ], limit=1)
+                ('start_date', '<=', fields.Datetime.to_string(request.date)),
```

Comment:
Oops, this shouldn't be put inside to_string, because request.date is already a string!

Reply:
In the fields of sms_child_request, date is a fields.Datetime object.. request is not a sms_child_request ?